### ISSUE TITLE
Parameter to pass the subscriber subaccount for scenarios without user token

### ISSUE THREAD

**Issue by ElectronicWizzard** [SAP/cloud-sdk-js#564]
**Is your feature request related to a problem? Please describe.**

We are using the Cloud SDK for Javascript Destination API in background processes. In this scenario we do not have a user JWT from which the destination service can derive the subsrciber subaccount required as we are running in a multitenancy scenario.
Currently we are using a inoffical workaround
```
//This is just a workaround to read from subscriber subaccount "ccsetuppocsubsriber" without a user token be creating the following dummy url.
getDestinationFromDestinationService("xxxx", { iss: 'https://ccsetuppocsubsriber.example.com' } ).then( value => {
    console.log("Destination from subscriber tenant"  );
    console.log(value.url);    
} ).catch( rejected => {
    console.log("xxxx not found in subscriber tenant" );
});
```
**Describe the solution you'd like**
We would rather aspect a parameter in the API which allows to pass the subaccount which is as well documented as probably we are not the only consumer running it as part of a background job. Using a not documented feature has as well the risk that we can not rely on that it will always work

**Describe alternatives you've considered**

Stay with the workaround hoping it will not break

**Comment by artemkovalyov** [comment]
@ElectronicWizzard, thanks for reporting this. @FrankEssenberger, can you look into this so that we can quickly discuss it.

**Comment by FrankEssenberger** [comment]
Hello ElectronicWizzard,

first of all I can ensure you that we always try to introduce no breaking changes in behavior on exported methods. Since the `getDestinationFromDestinationService` is public and the possibility to hand in the issuer `iss`  by hand will stay for all versions 1.X.Y. So the planned version 2.0 this may change.

The reason we did not expose this option more prominently with a seperate API is the potential danger of doing mistakes. In the JWT case the token is validated and only then a call to the destination service is done on behalf of the tenant indicated by the token (issuer field). If you allow the option for passing the issuer directly without a validation you really have to be super careful, that you do not make a mistake and set a wrong value and break the tenant isolation.

I will discuss with my colleagues what is their opinion on a method parameter as you describe it. I can see the point for processes without a tenant context (like background jobs).

Best
Frank

**Comment by marikaner** [comment]
@ElectronicWizzard, please note that this repository has been moved and the link to this issue has changed.

**Comment by FrankEssenberger** [comment]
https://github.com/SAP/cloud-sdk-js/pull/897

**Comment by FrankEssenberger** [comment]
We have properly documented the option so it is not a work around anymore, but a valid usecase. It will be published in the next release 1.3.6 or 1.4.0.
